azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/handlers/CallStateHandler.kt

```diff
@@ -34,10 +40,20 @@ internal class CallStateHandler(
         return store.getStateFlow().value.callState.callingStatus.callCompositeCallState()
     }
 
-    private fun sendCallStateChangedEvent(status: CallingStatus) {
+    private fun sendCallStateChangedEvent(
+        status: CallingStatus,
+        callEndReasonCode: Int?,
+        callEndReasonSubCode: Int?
+    ) {
         try {
```

Comment:
Try catch is too open here, should be inside the forEach.

I.e. if one event fails, it'll cease propagation to the rest.

should be 
forEach {
 event = CallCompositeCallStateChangedEvent..
 try { it.handle(event) } catch (e) { /* do nothing */ }
 
 to make sure that an error in one handler doesn't affect others.
}